### DISCUSSION TITLE
How to catch publish ack error

### DISCUSSION THREAD

**Discussion by lyw1991** [mqttjs/MQTT.js#1811] [Q&A]
When I publish message to a topic without acl, how can I catch the acl error?
The publish callback return like:
`{ cmd: 'pubrel', qos: 2, messageId: 30411 }`

MQTTX client can catch the error
![image](https://github.com/mqttjs/MQTT.js/assets/18585437/a03cb550-e0d0-4084-a095-45368fe7f3e7)

**Comment by robertsLando** [comment]
@lyw1991 

```js
client.pulbish(...., (err, packet) => {
	if(err.code === 5) // not authorized
})
```

You can get all reason codes messages by using `ReasonCodes` enum:

```ts
import { ReasonCodes } from 'mqtt'
```
